build.sh

```diff
@@ -89,6 +89,13 @@ build_fcct() {
     rm fcct -rf
 }
 
+configure_tang() {
+    install -m 0755 -T "$srcdir"/src/tang/tangdw /usr/libexec/tangdw
+    install -m 0644 -T "$srcdir"/src/tang/tangd.xinetd /etc/xinetd.d/tangd
+    /usr/libexec/tangd-keygen /var/db/tang
```

Comment:
I think this and the line below should be done *dynamically* when we start the test, not as part of the build.

Doing it here for example means that every container build has the same key.

This type of thing is why OSTree systems have a hard requirement that `/var` starts out empty...it doesn't matter for our container images but ideally the way we ship code in containers matches how we ship them for the host too.

Reply:
How important is it to have a different tang key for testing?  I was actually thinking of eventually using a static set of keys to facilitate converting these to external tests.

Reply:
How important is it to have a different tang key for testing?  I was actually thinking of eventually using a static set of keys to facilitate converting these to external tests.

Reply:
The external tests we've written so far are generally "static", but nothing stops us from *generating* them - that's what https://github.com/ostreedev/ostree/pull/2048 is starting to do (the binary is generated, not committed to git).

So what you want is to do something like dynamically generate the Ignition that would contain the key, rather than commit it to git?  We could add something like `make kola-tests` to fedora-coreos-config that generates subdirectories in `tests/kola`.

Or, we could ship tests as container images (but just unpack them as fancy tarballs rather than running them) or, ship them as RPMs or tarballs put up somewhere.

---

Or, if you want to keep using a static key, just move the directory from `/var/db/tang` to `/usr/lib/coreos-assembler/tang` or so.  That addresses my objections with storing static data in `/var`.

Reply:
I moved the key generation into the test.